### ISSUE TITLE
can't support pattern /src/*/index.*

### ISSUE THREAD

**Issue by tianyingchun** [mrmlnc/fast-glob#266]
### Environment

  * OS Version: mac os 10.14.5
  * Node.js Version: v13.12.0

### Actual behavior
```
input `/src/*/index.*` ==>  "src/*/index.*"
```
### Expected behavior
input `/src/*/index.*` ==>  ["src/A/index.ts", "src/B/index.ts"]
### Steps to reproduce

  1. …

### Code sample

```js
// Paste your code here.
```

**Comment by papb** [comment]
Shouldn't you be using `/src/**/index.*` instead?

**Comment by mrmlnc** [comment]
Looks like a bug, because the `deep` filter ignore nested directories for patterns that starts with `/`. Most likely this is related to the [`partial`](https://github.com/mrmlnc/fast-glob/blob/master/src/providers/matchers/partial.ts#L5) matcher.

As workaround you can use `cwd` option:

```ts
const fg = require('fast-glob');

const entries = fg.sync('lib/*/libpt*', {
	cwd: '/'
});

console.dir(entries, { colors: true });
```

**Comment by mrmlnc** [comment]
Shipped with [`3.2.3`](https://github.com/mrmlnc/fast-glob/releases/tag/3.2.3).
